src/main/java/org/folio/services/transactions/TransactionApiService.java

```diff
@@ -84,7 +85,21 @@ public Future<Transaction> createAllocation(Transaction allocation, RequestConte
     return processBatch(batch, requestContext)
       .map(v -> allocation)
       .onSuccess(v -> log.info("Success creating allocation, id={}", allocation.getId()))
-      .onFailure(t -> log.error("Error creating allocation, id={}", allocation.getId(), t));
+      .recover(t -> {
+        log.error("Error creating allocation, id={}", allocation.getId(), t);
+        if (t instanceof HttpException httpException) {
+          var errorCode = httpException.getErrors().getErrors().get(0).getCode();
```

Comment:
some http exceptions can not have errors populated, so need to add check here

Reply:
If not we should add some utility method that can handle all error codes that BatchTransactionChecks can provide.
Also to avoid multiple names we can use exactly the same error code names as mod-finance-storage has

Reply:
> probably we need to make more work here, BatchTransactionChecks in mod-finance-storage can throw multiple error codes, do we send them in mod-finance response or not. For example, did we send "budgetRestrictedExpendituresError" before this change?


based on previous implementation, all error codes come from mod-finance-storage, forwarded to response. example response for budgetRestrictedExpendituresError
![image](https://github.com/user-attachments/assets/464b2d95-9d91-4e35-b3b6-5d7ec41c0faa)

Reply:
Yeah, I agree. There is detailed error code for UI that can be handled based on context. I will mention in the task. Thanks